### ISSUE TITLE
[VSCode] Suggest alternative names during rename operations

### ISSUE THREAD

**Issue by Sma1lboy** [TabbyML/tabby#3155]
**Please describe the feature you want**
When renaming variables, functions, or other symbols in VSCode, provide a list of contextually appropriate alternative names to choose from.

<img width="294" alt="image" src="https://github.com/user-attachments/assets/5c38e977-4910-4fa0-9531-247a894ff4a1">

---
Please reply with a 👍 if you want this feature.

**Comment by qiyx2019** [comment]
How is this function implemented? Is there a demo?

**Comment by Sma1lboy** [comment]
@qiyx2019 It's from GitHub Copilot. I'm not sure if they are using any private VSCode APIs. I'll provide more information later.
Here's a little demo of how it looks.

https://github.com/user-attachments/assets/174c217e-148e-47ac-a367-6dc44143c94b

**Comment by qiyx2019** [comment]
yes ,i want know this input is how to implemented? do you have any good ideas?
